src/formatter/builder.ts

```diff
@@ -99,28 +110,43 @@ const FormatterBuilder = {
     let Syntax = JavascriptSnippetSyntax
     if (doesHaveValue(snippetSyntax)) {
       const fullSyntaxPath = path.resolve(cwd, snippetSyntax)
-      Syntax = require(fullSyntaxPath) // eslint-disable-line @typescript-eslint/no-var-requires
+      Syntax = await importer(fullSyntaxPath, true)
+      Syntax = FormatterBuilder.resolveConstructor(Syntax)
     }
     return new StepDefinitionSnippetBuilder({
       snippetSyntax: new Syntax(snippetInterface),
       parameterTypeRegistry: supportCodeLibrary.parameterTypeRegistry,
     })
   },
 
-  loadCustomFormatter(customFormatterPath: string, cwd: string) {
-    const CustomFormatter = createRequire(cwd)(customFormatterPath)
```

Comment:
Unfortunately had to add some fragmentation back here, as the ESM equivalent of `createRequire` is [not yet stable](https://nodejs.org/api/esm.html#esm_import_meta_resolve_specifier_parent).